include/cuda/std/detail/libcxx/include/numeric

```diff
@@ -16,6 +16,7 @@
 namespace std
 {
 
+#ifndef __cuda_std__
```

Comment:
Note to self, we should rather pull out the relevant pieces that we need (looking at https://github.com/llvm/llvm-project/tree/main/libcxx/include/__numeric)

That said, no chage requested for now